tests/PHPUnit/Domain/Service/Precondition/BeginnerPreconditionsUnitTest.php

```diff
@@ -0,0 +1,65 @@
+<?php declare(strict_types=1);
+
+namespace PhpTuf\ComposerStager\Tests\PHPUnit\Domain\Service\Precondition;
+
+use PhpTuf\ComposerStager\Domain\Service\Precondition\BeginnerPreconditions;
+use PhpTuf\ComposerStager\Domain\Service\Precondition\CommonPreconditionsInterface;
+use PhpTuf\ComposerStager\Domain\Service\Precondition\StagingDirDoesNotExistInterface;
+use PhpTuf\ComposerStager\Domain\Value\Path\PathInterface;
+use PhpTuf\ComposerStager\Tests\PHPUnit\TestCase;
+
+/**
+ * @coversDefaultClass \PhpTuf\ComposerStager\Domain\Service\Precondition\BeginnerPreconditions
+ *
+ * @uses \PhpTuf\ComposerStager\Domain\Service\Precondition\AbstractPrecondition
+ *
+ * @property \PhpTuf\ComposerStager\Domain\Service\Precondition\CommonPreconditionsInterface|\Prophecy\Prophecy\ObjectProphecy $commonPreconditions
+ * @property \PhpTuf\ComposerStager\Domain\Service\Precondition\StagingDirDoesNotExistInterface|\Prophecy\Prophecy\ObjectProphecy $stagingDirDoesNotExist
+ * @property \PhpTuf\ComposerStager\Domain\Value\Path\PathInterface|\Prophecy\Prophecy\ObjectProphecy $activeDir
+ * @property \PhpTuf\ComposerStager\Domain\Value\Path\PathInterface|\Prophecy\Prophecy\ObjectProphecy $stagingDir
+ */
+class BeginnerPreconditionsUnitTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        $this->activeDir = $this->prophesize(PathInterface::class);
+        $this->activeDir
+            ->resolve()
+            ->willReturn(self::ACTIVE_DIR);
+        $this->stagingDir = $this->prophesize(PathInterface::class);
+        $this->stagingDir
+            ->resolve()
+            ->willReturn(self::STAGING_DIR);
+        $this->commonPreconditions = $this->prophesize(CommonPreconditionsInterface::class);
+        $this->stagingDirDoesNotExist = $this->prophesize(StagingDirDoesNotExistInterface::class);
+    }
+
+    protected function createSut(): BeginnerPreconditions
+    {
+        $commonPreconditions = $this->commonPreconditions->reveal();
+        $stagingDirDoesNotExist = $this->stagingDirDoesNotExist->reveal();
+        return new BeginnerPreconditions($commonPreconditions, $stagingDirDoesNotExist);
+    }
+
+    /**
+     * @covers ::__construct
+     * @covers ::isFulfilled
+     */
+    public function testIsFulfilled(): void
+    {
+        $activeDir = $this->activeDir->reveal();
```

Comment:
Is there a reason why this test does not have a data provider to see what happens if one of the preconditions returns false?

Reply:
Well, yes. The reason is that that behavior happens in `AbstractPrecondition`, which is already covered by its unit test. But I suppose it is _technically_ a risk... I'll add the test here, too.